tests/adapters_t/batchsystems_t/test_slurm.py

```diff
@@ -0,0 +1,212 @@
+            "scontrol update NodeName=host-10-18-1-1 State=DRAIN Reason='COBalD/TARDIS'"
+        )
+
+    def test_integrate_machine(self):
+        self.assertIsNone(
+            run_async(self.slurm_adapter.integrate_machine, drone_uuid="VM-1")
+        )
+
+    def test_get_resource_ratios(self):
+        self.assertEqual(
+            list(run_async(self.slurm_adapter.get_resource_ratios, drone_uuid="VM-1")),
+            [self.cpu_ratio, self.memory_ratio],
+        )
+        self.mock_async_run_command.assert_called_with(self.command)
+
+        self.assertEqual(
+            run_async(self.slurm_adapter.get_resource_ratios, drone_uuid="not_exists"),
+            {},
+        )
+
+    def test_get_resource_ratios_without_options(self):
+        self.setup_config_mock()
+        del self.config.BatchSystem.options
+        self.slurm_adapter = SlurmAdapter()
+
+        self.assertEqual(
+            list(run_async(self.slurm_adapter.get_resource_ratios, drone_uuid="VM-1")),
+            [self.cpu_ratio, self.memory_ratio],
+        )
+
+        self.mock_async_run_command.assert_called_with(self.command_wo_options)
+
+    def test_get_allocation(self):
+        self.assertEqual(
+            run_async(self.slurm_adapter.get_allocation, drone_uuid="VM-1"),
+            max([self.cpu_ratio, self.memory_ratio]),
+        )
+        self.mock_async_run_command.assert_called_with(self.command)
+
+        self.assertEqual(
+            run_async(self.slurm_adapter.get_allocation, drone_uuid="not_exists"),
+            0.0,
+        )
+
+    def test_get_machine_status(self):
+        state_mapping = {
+            "VM-1": MachineStatus.Available,
+            "not_exists": MachineStatus.NotAvailable,
+            "draining_m": MachineStatus.Draining,
+            "idle_m": MachineStatus.Available,
+            "drained_m": MachineStatus.NotAvailable,
+            "pwr_up_m": MachineStatus.NotAvailable,
+        }
+
+        for machine, state in state_mapping.items():
+            self.assertEqual(
+                run_async(self.slurm_adapter.get_machine_status, drone_uuid=machine),
+                state,
+            )
+
+        self.mock_async_run_command.reset_mock()
+
+        self.mock_async_run_command.side_effect = CommandExecutionFailure(
+            message="Test", exit_code=123, stderr="Test"
+        )
+        with self.assertLogs(level="WARN"):
+            with self.assertRaises(CommandExecutionFailure):
+                attributes = {
+                    "Machine": "Machine",
+                    "State": "State",
+                    "Activity": "Activity",
+                    "TardisDroneUuid": "TardisDroneUuid",
+                }
+                # Escape Slurm expressions and add them to attributes
+                attributes.update(
+                    {
+                        key: quote(value)
+                        for key, value in self.config.BatchSystem.ratios.items()
```

Comment:
`self.config.BatchSystem.ratios` is not set anywhere in this unittest. In addition, it nothing that can be set according to the documentation. I would suggest to remove it then. Or am I wrong here?

Reply:
This is most likely a copy paste remnant where I adapted the comment but forgot to think about the code ;)